cdap-proto/src/main/java/co/cask/cdap/proto/metadata/lineage/ProgramRecord.java

```diff
@@ -16,17 +16,17 @@
 
 package co.cask.cdap.proto.metadata.lineage;
 
-import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.id.NamespacedEntityId;
 
 import java.util.Objects;
 
 /**
  * Class to serialize program in {@link LineageRecord}.
  */
 public class ProgramRecord {
-  private final Id.NamespacedId entityId;
+  private final NamespacedEntityId entityId;
```

Comment:
This may be ok, because we use a codec here. However, I think we should change the response in 4.0 to make it better, and provide release notes. It looks like the candidates are the metadata/lineage and usage registry APIs.